start.py

```diff
@@ -21,6 +21,31 @@
 admin_port = runtime_port + 1
 
 
+def x_frame_opt_headers_handler(x_frame_opt, x_frame_opt_uri):
+    base = "add_header X-Frame-Options "
+
+    if x_frame_opt_uri:
+        return base + "\"ALLOW-FROM " + x_frame_opt_uri + "\";"
+
+    if x_frame_opt:
+        if x_frame_opt == "Never allow":
+            return base + "\"DENY\";"
+        elif x_frame_opt == "Allow":
+            # if it is Allow, than there shouldn't be a
+            # X-Frame-Options header in the first place,
+            # so replace line with empty string
+            return ""
+        elif x_frame_opt == "Allow on same domain":
+            return base + "\"SAMEORIGIN\";"
+    else:
+        # defaulting to allow all
+        return ""
+
+x_frame_opt = os.environ.get('X_FRAME_OPTIONS')
```

Comment:
one variable would be my preference, so:
    os.environ.get('X_FRAME_OPTIONS')
the value would contain the literal header value, and if None or 'Allow' it would not set any header.